src/utility/Sd2Card.cpp

```diff
@@ -248,7 +248,7 @@ uint8_t Sd2Card::init(uint8_t sckRateID, uint8_t chipSelectPin) {
   errorCode_ = inBlock_ = partialBlockRead_ = type_ = 0;
   chipSelectPin_ = chipSelectPin;
   // 16-bit init start time allows over a minute
-  uint16_t t0 = (uint16_t)millis();
+  unsigned int t0 = millis();

```

Comment:
I don't like this one. millis() returns a 32 bit value. We should only use the unsigned int for the delta AFTER having done the subtraction in 32 bits. Notice that, AVR has unsigned long millis(), SAMD has uint32_t millis()

Reply:
> We should only use the unsigned int for the delta AFTER having done the subtraction in 32 bits

why?

> Notice that, AVR has unsigned long millis(), SAMD has uint32_t millis()

SAMD return `unsigned long` as AVR: https://github.com/arduino/ArduinoCore-samd/blob/master/cores/arduino/delay.h#L36